ports/samd/modsamd.c

```diff
@@ -36,12 +36,13 @@ extern const mp_obj_type_t samd_flash_type;
 
 STATIC mp_obj_t samd_pininfo(mp_obj_t pin_obj) {
     const machine_pin_obj_t *pin_af = pin_find(pin_obj);
-    // Get the name, now that it is not in the pin object
-    const char *name = pin_af->name;
+    // Get the name, now that it is in the pin object
+    size_t len;
+    const char *name = (char *)qstr_data(pin_af->name, &len);
 
     #if defined(MCU_SAMD21)
     mp_obj_t tuple[7] = {
-        tuple[0] = mp_obj_new_str(name, strlen(name)),
+        tuple[0] = mp_obj_new_str(name, len),
```

Comment:
Can this be `tuple[0] = MP_OBJ_NEW_QSTR(pin_af->name)`?  That's much simpler.

Reply:
Thanks for the hint. I had the impression that there was an easier way.